juniper_official/Interfaces/logical-interface-in-traffic.rule

```diff
@@ -0,0 +1,192 @@
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "$interface-name.$sub-interface-index input traffic is above high threshold($high-threshold octets)";
+                        }
+                    }
+                }
+                /*
+                 * Sets color to yellow and sends out an anomaly notification
+                 * when the interface input traffic($in-octets-stats-value) is
+                 * above low threshold (in_octets_low_threshold) for all points
+                 * in a 60 seconds period.
+                 */
+                term is-interface-in-traffic-median {
+                    when {
+                        min-rate-of-increase "$in-octets-stats-value" {
+                            rate "$low-threshold";
+                            time-range 5m;
+                        }
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "$interface-name.$sub-interface-index input traffic is in medium range(>=$low-threshold,<=$high-threshold  octets)";
+                        }
+                    }
+                }
+                /*
+                 * Defaults color to green.
+                 */
+                term interface-in-traffic-normal {
+                    then {
+                        status {
+                            color green;
+                            message "$interface-name.$sub-interface-index input traffic:$in-octets-stats-value is normal";
+                        }
+                    }
+                }
+            }
+            /*
+             * Variables with default values. Default values can be changed
+             * while deploying playbook instance.
+             */
+            variable in_octets_high_threshold {
+                value 80000000;
+                description "Sub Interface input traffic high threshold value in octets. Traffic rate of increase between metrics, before we report anomaly. Default value is 80000000 octets (80 MB)";
+                type int;
+            }
+            variable in_octets_low_threshold {
+                value 50000000;
+                description "Sub Interface input traffic low threshold value in octets. Traffic rate of increase between metrics, before we report anomaly. Default value is 50000000 octets (50 MB)";
+                type int;
+            }
+            variable stats_name {
+                value in-octets;
+                description "Sub Interface input traffic counter name";
+                type string;
+            }
+            variable ifd_name {
+                value ".*";
+                description "Interfaces to be monitored, regular expression, eg 'ge-.*'";
+                type string;
+            }
+            variable ifl_no {
```

Comment:
Generic comment: Please don't use underscores for any of the variables.

Reply:
Relaced under scores with hyphen

Reply:
Replaced under scores with hyphen and committed code